subsys/pm/device_runtime.c

```diff
@@ -24,6 +25,31 @@ LOG_MODULE_DECLARE(pm_device, CONFIG_PM_DEVICE_LOG_LEVEL);
 
 #define EVENT_MASK		(EVENT_STATE_ACTIVE | EVENT_STATE_SUSPENDED)
 
+static int invoke_suspend(const struct device *dev)
+{
+	pm_policy_device_power_lock_put(dev);
+	pm_device_domain_children_turn_off(dev);
+
+	if (dev->pm_base->action_cb(dev, PM_DEVICE_ACTION_SUSPEND) < 0) {
+		pm_device_domain_children_turn_on(dev);
+		pm_policy_device_power_lock_get(dev);
+		return -EAGAIN;
```

Comment:
This also overrides return values from the drivers, which was not previously the case

Reply:
return value now returned